python/dgl/sparse/sparse_matrix.py

```diff
@@ -575,7 +575,11 @@ def range_select(self, dim: int, index: slice):
         if dim not in (0, 1):
             raise ValueError("The selection dimension should be 0 or 1.")
         if isinstance(index, slice):
-            raise NotImplementedError
+            start = 0 if index.start is None else index.start
+            end = index.stop
```

Comment:
if we don't support step, raise error here.

Reply:
I think it is better to raise it when users input step is not 1 or None.

Reply:
SGTM.